include/lemlib/chassis/differential/motions/motion.hpp

```diff
@@ -0,0 +1,67 @@
+#pragma once
+
+#include "lemlib/chassis/differential/drivetrain.hpp"
+
+namespace lemlib::differential {
+/**
+ * @brief abstract motion class
+ *
+ * This is an abstract motion class for a differential drivetrain. It can be used to create
+ * custom motions for the chassis, implemented as part of LemLib, part of an extension for
+ * LemLib, or a user made extension specific to a single project.
+ *
+ * @b Example Child Class
+ * @code {.cpp}
+ * class SimpleForward : public Motion {
+ *   public:
+ *     SimpleForward() {}
+ *
+ *     void calculate(Drivetrain& drivetrain, const ChassisState& state) override {
+ *       drivetrain.left.move(127);
+ *       drivetrain.right.move(127);
+ *     }
+ *
+ *     bool isFinished() override {
+ *       return false; // this motion never finishes
+ *     }
+ * };
+ * @endcode
+ */
+class Motion {
+    public:
+        /**
+         * @brief calculate and apply new outputs for the drivetrain
+         *
+         * @param drivetrain a reference to the drivetrain. This is passed here instead of the ctor to
+         * simplify the creation of the motion, and to discourage the use of the drivetrain outside of
+         * the calculate method
+         * @param state the current position, velocity, and acceleration of the chassis
+         *
+         * @b Example
```

Comment:
user code should never interact with motion objects, so is there a point to providing an example?

Reply:
Why not? But we can add a note specifying that it is unlikely for anything outside of the chassis handler (or whatever we end up calling it) to call it. Adding example code here can provide additional context and I think it will be beneficial

Reply:
actually upon further analysis, yea it doesn't make sense to provide example code for this